OpenSim/Simulation/Wrap/WrapEllipsoid.cpp

```diff
@@ -1294,3 +1296,25 @@ double WrapEllipsoid::closestPointToEllipse(double a, double b, double u,
 
     return sqrt(dx*dx + dy*dy);
 }
+// Implement generateDecorations by WrapEllipsoid to replace the previous out of place implementation 
+// in ModelVisualizer
+void WrapEllipsoid::generateDecorations(bool fixed, const ModelDisplayHints& hints, const SimTK::State& state,
+    SimTK::Array_<SimTK::DecorativeGeometry>& appendToThis) const 
+{
+
+    Super::generateDecorations(fixed, hints, state, appendToThis);
+    if (fixed) return;
+
+    if (hints.get_show_wrap_geometry()) {
+        const Vec3 color(SimTK::Cyan);
+        SimTK::Transform ztoy;
+        ztoy.updR().setRotationFromAngleAboutX(SimTK_PI / 2);
```

Comment:
I understand the reason for rotating the Cylinder but why the Ellipsoid? Are the radii for the WrapEllipsoid (rx, ry, rz)  not consistent with the DecorativeEllipsoid radii? The old ModelVisualizer code was not doing this either, which makes think this might be copy and paste bug.